### PR TITLE
Add packaging checks

### PR DESCRIPTION
**Pull request by Delaunay**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
 
-on: [push, pull_request]
+on:
+  pull_request:
+    branches:
+      - master
+      - develop
+  release:
+    types:
+      [published]
```
Comment (bouthilx) [inline .github/workflows/test.yml:10]: It's nice to have `workflow_dispatch` as well to be able start tests when necessary.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -32,3 +39,39 @@ jobs:
...
+
+  pypi:
+    needs: [test]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
```
Comment (bouthilx) [inline .github/workflows/test.yml:53]: I think we should test on all versions the package supports, ideally 3.6 to 3.9 to be aligned with Oríon.

Comment (Delaunay) [reply .github/workflows/test.yml:53]: You commented on the python version that is [coming from Orion](https://github.com/Epistimio/orion/blob/develop/.github/workflows/build.yml#L159) :p

File: .github/workflows/test.yml (hunk 1 above)
Comment (Delaunay) [reply .github/workflows/test.yml:10]: added

File: .github/workflows/test.yml (hunk 2 above)
Comment (Delaunay) [reply .github/workflows/test.yml:53]: only 3.6 was missing, I added it

Comment (Delaunay) [reply .github/workflows/test.yml:53]: we have 3.10 too so when orion drops 3.6 by 3.10 we will be ready

**Pull request merged**